Objects/dictobject.c

```diff
@@ -1084,7 +1088,6 @@ insertdict(PyDictObject *mp, PyObject *key, Py_hash_t hash, PyObject *value)
     if (ix == DKIX_ERROR)
         goto Fail;
 
-    assert(PyUnicode_CheckExact(key) || mp->ma_keys->dk_lookup == lookdict);
```

Comment:
Maybe put this assertion after `MAYBE_NONUNICODE` when it will be true.

Reply:
I couldn't find a good place for this assertion. After `MAYBE_NONUNICODE`, the condition is obviously true.